### PR TITLE
fix: ApisixRoute update operations is not effective

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#312 

- What does this PR do:
  * Unify the route, upstream synchronization to the `manifest`, it's used by `ApisixRouteController`, `IngressController` and `EndpointController`.
  * Fixed the route synchronization bug.
  * Remove useless codes in `pkg/seven/state`.
  * Modify the e2e case `Create and remove the ApisixRoute`, adding update operation and sending requests to APISIX.

### CODE REVIEW

**Comment by tokers** [comment]
@gxthrj Please take a look when you have time.

File: test/e2e/ingress/resourcepushing.go [hunk 1]
```diff
@@ -99,14 +98,56 @@ spec:
...
+
+		// update
+		apisixRoute = fmt.Sprintf(`
+apiVersion: apisix.apache.org/v2alpha1
+kind: ApisixRoute
+metadata:
+  name: httpbin-route
+spec:
+  http:
+  - name: rule1
```
Comment (gxthrj) [inline test/e2e/ingress/resourcepushing.go:111]: should add cases : 
the diff `name` in the same `ApisixRoute`.
the same `name` in the diff `ApisixRoute`.

Comment (tokers) [reply test/e2e/ingress/resourcepushing.go:111]: Cannot capture your mean 100%.

Comment (gxthrj) [reply test/e2e/ingress/resourcepushing.go:111]: I mean add test cases:
1. In the same `ApisixRoute` yaml ,but change `spec.http.name` when update
2. In the different `ApisixRoute` yaml, but have the same `spec.http.name`, which means name is conflict.

Comment (tokers) [reply test/e2e/ingress/resourcepushing.go:111]: OK, will add it.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**